### PR TITLE
remove obsolete policies right after loading a game

### PR DESCRIPTION
**Pull request by Grummel7**
See #3774

### CODE REVIEW

File: Empire/Empire.cpp [hunk 1]
```diff
@@ -318,22 +318,6 @@ void Empire::AdoptPolicy(const std::string& name, const std::string& category,
...
-            m_adopted_policies.erase(policy_name);
-            continue;
-        }
-
-        if (adoption_info.category.empty()) {
-            ErrorLogger() << "UpdatePolicies found policy " << policy_name << " in empty category?";
-            m_adopted_policies.erase(policy_name);
-        }
-    }
-
```
Comment (Grummel7) [inline Empire/Empire.cpp:336]: As far as I understand it, doing these checks after loading, i.e. after a possible change in rules, should be enough. Not really sure about the sure about the second one, though.

File: server/SaveLoad.cpp [hunk 2]
```diff
@@ -490,6 +490,10 @@ void LoadGame(const std::string& filename, ServerSaveGameData& server_save_game_
             }
         }
 
+        for (auto it : empire_manager) {
```
Comment (geoffthemedio) [inline server/SaveLoad.cpp]: don't need the `{}` for a one-line loop body

File: Empire/Empire.cpp [hunk 3]
```diff
@@ -2783,6 +2767,35 @@ void Empire::UpdateOwnedObjectCounters(const Universe& universe) {
     }
 }
 
+
+void Empire::CheckNewGameRules()
```
Comment (geoffthemedio) [inline Empire/Empire.cpp]: misleading title... content isn't a "Rule"

File: server/SaveLoad.cpp (hunk 2 above)
Comment (Grummel7) [reply server/SaveLoad.cpp]: I am used to using them always, but I can adapt...

**Comment by geoffthemedio** [comment]
needs rebase

**Comment by Grummel7** [comment]
rebased

File: server/SaveLoad.cpp [hunk 4]
```diff
@@ -490,6 +490,9 @@ void LoadGame(const std::string& filename, ServerSaveGameData& server_save_game_
             }
         }
 
+        for (auto it : empire_manager)
```
Comment (geoffthemedio) [inline server/SaveLoad.cpp:493]:
```suggestion
        for (auto& entry : empire_manager)
```

File: server/SaveLoad.cpp [hunk 5]
```diff
@@ -490,6 +490,9 @@ void LoadGame(const std::string& filename, ServerSaveGameData& server_save_game_
...
+            it.second->CheckObsoleteGameContent();
```
Comment (geoffthemedio) [inline server/SaveLoad.cpp:494]:
```suggestion
            entry.second->CheckObsoleteGameContent();
```

File: Empire/Empire.cpp [hunk 6]
```diff
@@ -2783,6 +2767,35 @@ void Empire::UpdateOwnedObjectCounters(const Universe& universe) {
     }
 }
 
+
+void Empire::CheckObsoleteGameContent()
+{
```
Comment (geoffthemedio) [inline Empire/Empire.cpp]:
```suggestion
void Empire::CheckObsoleteGameContent() {
```

File: server/SaveLoad.cpp (hunk 5 above)
Comment (geoffthemedio) [inline server/SaveLoad.cpp]:
```suggestion
        for (auto& entry : empire_manager)
            entry.second->CheckObsoleteGameContent();
```

**Pull request merged**